lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,85 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SshAgent
+    attr_reader :pid
+    attr_reader :sock
+
+    def initialize(agent_socket, ssh_private_key)
+      FileUtils.mkdir_p(File.dirname(agent_socket))
```

Comment:
Can we make agent_socket optional? (from the man page of ssh-agent(1) "The default is $TMPDIR/ssh-XXXXXXXXXX/agent.<ppid>")